test/ec_internal_test.c

```diff
@@ -156,6 +156,53 @@ static int field_tests_ecp_mont(void)
 }
 
 #ifndef OPENSSL_NO_EC2M
+/* Test that decoding of invalid GF2m field parameters fails. */
+static int ec2m_field_sanity(void)
+{
+    BN_CTX *ctx = BN_CTX_new();
+    BIGNUM *p, *a, *b;
+    EC_GROUP *group1, *group2, *group3;
+
+    TEST_info("Testing GF2m hardening\n");
+
+    BN_CTX_start(ctx);
+    p = BN_CTX_get(ctx);
+    a = BN_CTX_get(ctx);
+    b = BN_CTX_get(ctx);
+    BN_one(a);
+    BN_one(b);
+
+    /* Even pentanomial value should be rejected */
+    BN_set_word(p, 0xf2);
+    group1 = EC_GROUP_new_curve_GF2m(p, a, b, ctx);
+
+    if (group1 != NULL)
```

Comment:
You might want to add a cleanup in the if condition where group1 is not NULL to prevent memory leaks.

Reply:
The groups are explicitly freed on lines 187–199.